src/app/clusters/network-commissioning/network-commissioning.cpp

```diff
@@ -396,6 +449,113 @@ void Instance::HandleAddOrUpdateWiFiNetwork(HandlerContext & ctx, const Commands
     }
 }
 
+#if CHIP_DEVICE_CONFIG_ENABLE_WIFI_PDC
+void Instance::HandleAddOrUpdateWiFiNetworkWithPDC(HandlerContext & ctx,
+                                                   const Commands::AddOrUpdateWiFiNetwork::DecodableType & req)
+{
+    // Credentials must be empty when configuring for PDC, it's only present to keep the command shape compatible.
+    if (!req.credentials.empty())
+    {
+        ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Protocols::InteractionModel::Status::InvalidCommand, "credentials");
+        return;
+    }
+
+    auto && networkIdentity = req.networkIdentity.Value(); // presence checked by caller
+    if (networkIdentity.size() > kMaxCHIPCompactNetworkIdentityLength ||
+        Credentials::ValidateChipNetworkIdentity(networkIdentity) != CHIP_NO_ERROR)
+    {
+        ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Protocols::InteractionModel::Status::InvalidCommand, "networkIdentity");
+        return;
+    }
+
+    if (req.clientIdentifier.HasValue() && req.clientIdentifier.Value().size() != CertificateKeyId::size())
+    {
+        ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Protocols::InteractionModel::Status::InvalidCommand, "clientIdentifier");
```

Comment:
Again, maybe the context string should say "clientIdentifier too long"... Not sure about the tradeoff between clarity and codesize here.

Reply:
Yeah not sure what the convention is... Most usages I've seen in the code don't seem to provide any context so I was keeping it terse.